Fix: #1190. Remove algorithm specific rules for content identifiers. (#1314)
httpwg/http-extensions: draft-ietf-httpbis-message-signatures.md

5. The signer creates an ordered list of content identifiers representing the message content and signature metadata to be covered by the signature, and assigns this list as the signature's Covered Content.
   * Each identifier MUST be one of those defined in Section 2.
   * This list MUST NOT be empty, as this would result in creating a signature over the empty string.

>    * Signers SHOULD include `*created` and `*request-target` in the list.

Reviewer: Suggest we make this more generic, especially since it's all advice (SHOULD) and not requirement:

```suggestion
   * Signers SHOULD include `*request-target` in the list.
   * Signers SHOULD include a date stamp, such as the `date` header or the `*created` field in the list.
```

5. The signer creates an ordered list of content identifiers representing the message content and signature metadata to be covered by the signature, and assigns this list as the signature's Covered Content.
   * Each identifier MUST be one of those defined in Section 2.
   * This list MUST NOT be empty, as this would result in creating a signature over the empty string.
   * Signers SHOULD include `*created` and `*request-target` in the list.

>    * Signers SHOULD include `date` and `*request-target` in the list.

Reviewer 2: ```suggestion
```

1. Signers may not have control over the `Date` header. (pre-signed messages, XmlHttpRequest callers, etc.)
2. The `*request-target` guidance is redundant with the line above.

The Signature Input is a US-ASCII string containing the content that will be signed.  To create it, the signer concatenates together entries for each identifier in the signature's Covered Content in the order it occurs in the list, with each entry separated by a newline `"\n"`.  An identifier's entry is a US-ASCII string consisting of the lowercased identifier followed with a colon `":"`, a space `" "`, and the identifier's canonicalized value (described below).

> 

Author: ```suggestion

If Covered Content contains `*created` and the signature's Creation Time is undefined an implementation MUST produce an error.
```

> If Covered Content contains `*created` and the signature's Creation Time is undefined or the signature's Algorithm name starts with `rsa`, `hmac`, or `ecdsa` an implementation MUST produce an error.

Reviewer 3: Should not remove this entirely. Implementations should still produce an error if the corresponding header for a signature parameter is not found.

Reviewer: Is the suggestion that this constraint be applied regardless of algorithm name?

I completely agree that this "algorithm name starts with" language is hugely problematic.

Author: @llorllale @jricher right! Both of you are correct! Fixing...

Reviewer 3: @jricher 

> Is the suggestion that this constraint be applied regardless of algorithm name?

Yes.

Author: Fixed!

> If Covered Content contains `*expires` and the signature does not have an Expiration Time or the signature's Algorithm name starts with `rsa`, `hmac`, or `ecdsa` an implementation MUST produce an error.

Reviewer 3: Same here - should error if not found

Author: thx @llorllale, I fixed it below: does it work for you?
